lib/fizzy/leb128.hpp

```diff
@@ -2,8 +2,18 @@
 
 namespace fizzy
 {
-inline constexpr uint64_t leb128_decode(const uint8_t*) noexcept
+inline constexpr uint64_t leb128u_decode(const uint8_t* input) noexcept
 {
-    return 0;
+    uint64_t result = 0;
+    uint8_t result_shift = 0;
+
+    for (auto byte = input;; ++byte, result_shift += 7)
+    {
+        result |= static_cast<uint64_t>(*byte & 0x7F) << result_shift;
```

Comment:
What if we run out of the input buffer (malformed encoding)? Should we care about it now?